### DISCUSSION TITLE
Best idiomatic way to have a per-app layer?

### DISCUSSION THREAD

**Discussion by kenNash** [evan-liu/karabiner.ts#148] [Q&A] [answered]
I'd like to activate a “Tab Switcher Mode” per-app, with `tab`.
I mean: in any app, `tab` should activate a layer with the key combos used for tab-switching in that particular app.
In other words, I'd like to be able to use, e.g., `tab+right_arrow` for `next tab` whatever it means in any app (where there are tabs).

**Comment by evan-liu** [comment]
There is no "best idiomatic way", but here is one option: 

```ts
layer('⇥').manipulators([
  withCondition(ifApp('^some.app.id$'))([
    // Some special mapping
  ]),

  // Pre/Next tab in most apps, put in the last
  {
    '←': toKey('[', '⌘⇧'),
    '→': toKey(']', '⌘⇧'),
  }
])
```

**Reply by kenNash** [reply]
Thank you @evan-liu.
What if I want to put together different apps? This doesn't work (frankly, I didn't expect it would have…):
```
withCondition(
        ifApp('^com.google.Chrome$') || // For the first app put here it works
        ifApp('^com.apple.Safari$').        // for the second one it doesn't
      )
```

**Reply by evan-liu** [reply, marked as answer]
`ifApp('RegExp')` takes a [RegExp](https://regexr.com/) string, so you can try to make it match all the apps you want to cover.

**Reply by kenNash** [reply]
Great! 🙏
